File: scratch/validate_syntax.js
const fs = require('fs');
const path = require('path');

function validateJson(filePath) {
  try {
    const data = fs.readFileSync(filePath, 'utf8');
    const parsed = JSON.parse(data);
    console.log(`✅ ${path.basename(filePath)} is valid JSON! Contains ${parsed.length} entries.`);
    return true;
  } catch (e) {
    console.error(`❌ Error parsing ${path.basename(filePath)}:`, e.message);
    return false;
  }
}

const categoriesPath = path.join(__dirname, '../data/categories.json');
const promptsPath = path.join(__dirname, '../data/prompts.json');

const catValid = validateJson(categoriesPath);
const promptsValid = validateJson(promptsPath);

if (catValid && promptsValid) {
  console.log("All data files validated successfully!");
  process.exit(0);
} else {
  console.error("Data validation failed!");
  process.exit(1);
}

File: scratch/validate_syntax.py
import json
import os

def validate_json(file_path):
    try:
        with open(file_path, "r", encoding="utf-8") as f:
            data = json.load(f)
        print(f"[OK] {os.path.basename(file_path)} is valid JSON! Contains {len(data)} entries.")
        return True
    except Exception as e:
        print(f"[ERROR] Error parsing {os.path.basename(file_path)}: {str(e)}")
        return False

base_dir = os.path.dirname(__file__)
categories_path = os.path.join(base_dir, "../data/categories.json")
prompts_path = os.path.join(base_dir, "../data/prompts.json")

cat_valid = validate_json(categories_path)
prompts_valid = validate_json(prompts_path)

if cat_valid and prompts_valid:
    print("All data files validated successfully!")
    exit(0)
else:
    print("Data validation failed!")
    exit(1)
